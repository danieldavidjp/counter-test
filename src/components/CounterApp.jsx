import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value);

	function handleAdd() {
		setCounter(counter + 1);
	}
	function handleSubtract() {
		if (counter > 0) {
			setCounter(counter - 1);
		}
	}

	function handleReset() {
		setCounter(0);
	}
	return (
		<>
			<h1>CounterApp</h1>
			<h2> {counter} </h2>
			<button onClick={handleAdd}>+1</button>
			<button onClick={handleSubtract}>-1</button>
			<button onClick={handleReset}>Reset</button>
		</>
	);
};

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
};
