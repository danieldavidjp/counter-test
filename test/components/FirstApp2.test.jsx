import { render, screen } from "@testing-library/react";
import { FirstApp } from "../../src/components/FirstApp";

describe("<FirstApp /> Test", () => {
	const title = "Hi";
	const subTitle = "This is a subtitle";
	test("should match with the snapshot", () => {
		const { container } = render(<FirstApp title={title} />);

		expect(container).toMatchSnapshot();
	});

	test('should show the message "Hi"', () => {
		render(<FirstApp title={title}></FirstApp>);

		expect(screen.getByText(title)).toBeTruthy();
	});

	test("should show the tittle in an h1", () => {
		render(<FirstApp title={title}></FirstApp>);
		expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(title);
	});

	test("should show the subtitle send by props", () => {
		render(<FirstApp title={title} subTitle={subTitle}></FirstApp>);

		expect(screen.getByText(subTitle)).toBeTruthy();
	});
});
