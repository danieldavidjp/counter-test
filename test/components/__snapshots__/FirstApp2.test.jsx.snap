// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FirstApp /> Test should match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hi
  </h1>
  <p>
    No subtitle
  </p>
  <p>
    Daniel
  </p>
</div>
`;
